dspace-server-webapp/src/test/java/org/dspace/app/rest/signposting/controller/LinksetRestControllerIT.java

```diff
@@ -0,0 +1,677 @@
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+            bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                    .withName("Bitstream")
+                    .withDescription("description")
+                    .withMimeType(bitstreamMimeType)
+                    .withReaderGroup(internalGroup)
+                    .build();
+        }
+        context.restoreAuthSystemState();
+
+        String uiUrl = configurationService.getProperty("dspace.ui.url");
+        getClient().perform(get("/signposting/links/" + bitstream.getID())
+                        .header("Accept", "application/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$",
+                        Matchers.hasSize(3)))
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/entities/publication/" + item.getID() + "' " +
+                        "&& @.rel == 'collection' " +
+                        "&& @.type == 'text/html')]").exists())
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/signposting/linksets/" + item.getID() + "' " +
+                        "&& @.rel == 'linkset' " +
+                        "&& @.type == 'application/linkset')]").exists())
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/signposting/linksets/" + item.getID() + "/json" +
+                        "' && @.rel == 'linkset' " +
+                        "&& @.type == 'application/linkset+json')]").exists());
+
+        DSpaceServicesFactory.getInstance().getConfigurationService().reloadConfig();
+        metadataAuthorityService.clearCache();
+        choiceAuthorityService.clearCache();
+    }
+
+    @Test
+    public void findTypedLinkForBitstreamUnderEmbargo() throws Exception {
+        String bitstreamContent = "ThisIsSomeDummyText";
+        String bitstreamMimeType = "text/plain";
+
+        context.turnOffAuthorisationSystem();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Item Test")
+                .withIssueDate("2017-10-17")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .build();
+
+        Bitstream bitstream = null;
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+            bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                    .withName("Bitstream")
+                    .withDescription("description")
+                    .withMimeType(bitstreamMimeType)
+                    .withEmbargoPeriod("6 months")
+                    .build();
+        }
+        context.restoreAuthSystemState();
+
+        String uiUrl = configurationService.getProperty("dspace.ui.url");
+        getClient().perform(get("/signposting/links/" + bitstream.getID())
+                        .header("Accept", "application/json"))
+                .andExpect(status().isOk())
```

Comment:
Similar to above this is embargoed and should return a 401 or 404

Reply:
Fixed